dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventsExportController.java

```diff
@@ -308,28 +308,25 @@ Page<ObjectNode> getEventChangeLogByUid(
     org.hisp.dhis.tracker.export.Page<EventChangeLog> changeLogs =
         eventChangeLogService.getEventChangeLog(uid, pageParams);
 
-    if (changeLogs.isHasNext()) {
-      changeLogs
-          .getPager()
-          .setNextPage(
-              request.getRequestURI()
-                  + String.format(
-                      "?page=%d&pageSize=%d",
-                      requestParams.getPage() + 1, requestParams.getPageSize()));
+    String nextPage = null;
+    if (changeLogs.isNext()) {
+      nextPage =
+          request.getRequestURI()
+              + String.format(
+                  "?page=%d&pageSize=%d", requestParams.getPage() + 1, requestParams.getPageSize());
     }
-    if (changeLogs.isHasPrevious()) {
-      changeLogs
-          .getPager()
-          .setPrevPage(
-              request.getRequestURI()
-                  + String.format(
-                      "?page=%d&pageSize=%d",
-                      requestParams.getPage() - 1, requestParams.getPageSize()));
+
+    String previousPage = null;
+    if (changeLogs.isPrev()) {
+      previousPage =
+          request.getRequestURI()
+              + String.format(
+                  "?page=%d&pageSize=%d", requestParams.getPage() - 1, requestParams.getPageSize());
     }
 
     List<ObjectNode> objectNodes =
         fieldFilterService.toObjectNodes(changeLogs.getItems(), requestParams.getFields());
 
-    return Page.withPager("changeLogs", objectNodes, changeLogs);
+    return Page.withPager("changeLogs", objectNodes, changeLogs, previousPage, nextPage);
```

Comment:
I think we can create a simpler API that takes care of doing the prev/nextPage transformation transparently. This way every controller will get that for free 🥳 .

Lets use the `export.Page.with` instance method described before in a service comment.

The reason we have to pass in `objectNodes` is that we filtered the `changeLogs.items`. So we cannot use the `changeLogs.items` in `Page.withPager`.

It could look like this in the end 😋 

```java
return Page.withPager("changeLogs", changeLogs.with(objectNodes), request);
```

We pass
* the JSON key under which the `changeLogs.items` should be returned
* the `export.Page` (here we reuse the pagination information but only replace the items as they were filtered; without resorting to mutations)
* and the request to fill in the `prev/nextPage` if `changeLogs.isPrev/isNext` is true